xonsh/ptk/shell.py

```diff
@@ -142,8 +142,6 @@ def rprompt_tokens(self, cli):
         prompt.
         """
         p = builtins.__xonsh_env__.get('RIGHT_PROMPT')
-        if len(p) == 0:
-            return []
```

Comment:
These lines should be added back, but you should make it `if isinstance(p, str) and len(p) == 0:`